pkg/klusterlet/clusterclaim/clusterclaimer.go

```diff
@@ -95,14 +97,48 @@ func init() {
 }
 
 type ClusterClaimer struct {
-	ClusterName    string
-	Product        string
-	Platform       string
-	HubClient      client.Client
-	KubeClient     kubernetes.Interface
-	ConfigV1Client openshiftclientset.Interface
-	OauthV1Client  openshiftoauthclientset.Interface
-	Mapper         meta.RESTMapper
+	ClusterName      string
+	Product          string
+	Platform         string
+	HubClient        client.Client
+	KubeClient       kubernetes.Interface
+	ConfigV1Client   openshiftclientset.Interface
+	OauthV1Client    openshiftoauthclientset.Interface
+	Mapper           meta.RESTMapper
+	managedclusterID string
+}
+
+func NewClusterClaimer(
+	clusterName string,
+	hubClient client.Client,
+	kubeClient kubernetes.Interface,
+	configV1Client openshiftclientset.Interface,
+	oauthV1Client openshiftoauthclientset.Interface,
+	mapper meta.RESTMapper,
+) (ClusterClaimer, error) {
+	var managedclusterID string
+	// If clusterclaim is not created, get UID of kube-system namespace
+	ns, err := kubeClient.CoreV1().Namespaces().Get(context.TODO(), "kube-system", metav1.GetOptions{})
+	if err != nil && !errors.IsForbidden(err) {
+		klog.Errorf("failed to get kube-system namespace. err:%v", err)
+		return ClusterClaimer{}, err
+	}
+	// Create a new uuid as the managedclusterID
+	if errors.IsForbidden(err) {
```

Comment:
this should be openshift cluster id at first, then kube-system id, then random id.

Reply:
Yes, we discussed the above process before, but I found we need to create clustercliamer first to get the openshiftID, let me see if I move these things before create.